petastorm/tests/test_decode_row.py

```diff
@@ -39,3 +41,50 @@ def test_can_not_decode():
 
     with pytest.raises(DecodeFieldError, match='matrix'):
         decode_row(row, MatrixSchema)
+
+
+def test_decode_numpy_scalar_when_codec_is_none():
+    """Decoding a row that has a field with the codec set to None. The type should be deduced automatically
```

Comment:
In the tests where the codec is being inferred, can we compare the codec inferred explicitly? For example, instantiate the expected `UnischemaField` and use the class's `__eq__`.

Reply:
Not sure what do you mean?

Reply:
Test `UnischemaField.codec` has the correct type